### PR TITLE
test: initial stab at trying Ava test framework

### PR DESCRIPTION
**Pull request by dtribble**
re #1335 

A spike to see how AVA test frameworks.  See #1335 for discussion.

Only ERTP tests were completely converted, though many tests required no additional changes to pass.

### CODE REVIEW

**Review by erights — COMMENTED** [review]
LATM --- Looks Awesome To Me. Very enthusiastic!

**Comment by erights** [comment]
> LATM --- Looks Awesome To Me. Very enthusiastic!

Just to be clear, this wasn't approval. It is happy enthusiasm.

File: packages/ERTP/test/unitTests/test-issuerObj.js [hunk 1]
```diff
@@ -183,126 +171,112 @@ test('issuer.burn', t => {
...
-      t.ok(
+      t.assert(
         amountMath.isEqual(burntBalance, amountMath.make(837)),
         `entire minted payment was burnt`,
       );
-      t.rejects(() => issuer.getAmountOf(payment1), /payment not found/);
+      return t.throwsAsync(() => issuer.getAmountOf(payment1), { message: /payment not found/ });
     })
     .catch(e => t.assert(false, e));
 });
```
Comment (Chris-Hibbert) [inline packages/ERTP/test/unitTests/test-issuerObj.js:184]: t.notThrows() would be good here.

**Review by Chris-Hibbert — COMMENTED** [review]
I like `t.notThrows()`. I think it may be the answer to the issuer I've been pointing to.

File: packages/ERTP/test/unitTests/test-issuerObj.js (hunk 1 above)
Comment (warner) [reply packages/ERTP/test/unitTests/test-issuerObj.js:184]: Would our rule be to put `t.throwsAsync` or `t.notThrows` (`t.notThrowsAsync`??) around every call that returns a Promise? I can't imagine us being ok with any potential rejection going unnoticed, so every promise would fall into one of:

* we expect a positive result and want to examine it: use `checkResult(await ..)` or `t.notThrows(..).then(checkResult)`
* we expect a positive result but don't care about the value: use `await ..` or `t.notThrows(..);`
* we expect a negative result (testing delibrary failure situations): use `t.throwsAsync(..);`

For sync stuff, I expect an exception that throws high enough to be noticed by the enclosing test case to flunk the test. As long as any lower-level exceptions cause a Promise to reject, and we don't drop any promises on the floor inside our libraries, and our test cases are on the lookout for rejections, then arbitrary exceptions in our libraries ought to make the test cases flunk too.

**Comment by michaelfig** [comment]
@dtribble this PR looks like it's been done by @warner's latest changes.  Closing, but feel free to reopen if you need it for reference.

**Pull request closed without merging**